sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/UseUriInsteadOfStringMethodOverloads.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public static void FetchResource(string uriString) { }
+        public static void FetchResource(Uri uri) { }
+
+        public static void FetchResource2(Uri uri, bool whatever) { }
+        public static void FetchResource2(string stringUri, bool whatever) { }
+
+        public static void FetchResource3(object stringUri, bool whatever) { }
+
+        public static void FetchResource4(string stringUri, bool whatever) { }
+//                                        ^^^^^^ Noncompliant {{Either change this parameter type to 'System.Uri' or provide an overload which takes a 'System.Uri' parameter.}}
```

Comment:
Does this work for constructors? If FxCop is handling constructors, we should do as well.